internal/commands/vaultsecrets/secrets/create.go

```diff
@@ -283,7 +278,88 @@ func createRun(opts *CreateOpts) error {
 			if err := opts.Output.Display(newRotatingSecretsDisplayer(true).PreviewRotatingSecrets(resp.Payload.Config)); err != nil {
 				return err
 			}
+
+		default:
+			return fmt.Errorf("unsupported rotating secret provider type")
+		}
+
+	case secretTypeDynamic:
+		sc, err := readConfigFile(opts)
+		if err != nil {
+			return fmt.Errorf("failed to process config file: %w", err)
+		}
+
+		missingFields := validateSecretConfig(sc)
+
+		if len(missingFields) > 0 {
+			return fmt.Errorf("missing required field(s) in the config file: %s", missingFields)
+		}
+
+		switch sc.Type {
+		case integrations.AWS:
+
+			missingDetails := validateDetails(sc.Details, AwsRequiredKeys)
```

Comment:
Curious if there has been any past discussion around performing validation to the level of detail (i.e., individual keys on provider-specific APIs) we have here. I would be okay with us dropping the CLI validation for fields at this level. I could imagine it being difficult to keep the API and CLI in sync over time as fields are added / removed / made optional. The APIs typically return an error for any required fields that are missing which can be raised to users here. Lmk your thoughts!

Reply:
This is a great point. In the RFC, we added [a section](https://hermes.hashicorp.services/document/1MR6Fu60VISKJCBJhCpyEC1VqeizJCwJeiYbX6X88bd0) for a backend endpoint that would not only help users compose the configuration files, but also give us a source of truth to work from that isn't hard-coded "required keys" in this repo. We bumped that work's priority up to Milestone 3 of this project (instead of just being "future work"! While it's still hard-coded values though, we'll run into issues places like [here](https://github.com/hashicorp/hcp/pull/143/files#diff-7f92b17a95650abeca3ec432e8bcf6859f19709d1bdb6b3687fb5d704940a4ecR324) with panics on interface conversion when the value wasn't provided in the config file (since it would be nil and not a string atp).